### DISCUSSION TITLE
Power Platform CLI: output setting

### DISCUSSION THREAD

**Discussion by Laskewitz** [microsoft/powerplatform-build-tools#463] [Ideas]
I would love to have a user setting for output, so that you can easily switch between markdown or json output. This would also change the need for having to add `--json` to all commands that offer json output.

**Comment by appieschot** [comment]
Totally this! In CLI for Microsoft 365 we opted for a `--output` property that allows us to support `json`, `md`, `csv` and `text`. Working with CLI's there are a lot of times you want to work with `json` but for quite a few listing scenarios life is easier if you can output to `csv` or `md` and quickly printing when you are working with a colleague the `text` output is a life saver! 

Totally has my upvotes 👍

**Reply by appieschot** [reply]
Oh and in addition it would be really awesome if you would be able to set a default output mode as a global setting preventing one to keep adding `--output json` to each command when working with more complex scripts.

**Comment by vignaesh01** [comment]
Definitely need json output format to write toolling on top of PAC CLI. It is very difficult to parse tabular fixed width responses.
